src/media/js/elements/header.js

```diff
@@ -44,23 +48,50 @@ define('elements/header',
                     var headerChildren = root.querySelectorAll(
                         'mkt-header-child');
                     for (var i = 0; i < headerChildren.length; i++) {
-                        headerChildren[i].header = root;
-                        root.parentNode.insertBefore(headerChildren[i],
-                                                     root.nextSibling);
+                        if (root.nextSibling) {
+                            root.parentNode.insertBefore(headerChildren[i],
+                                                         root.nextSibling);
+                        } else {
+                            root.parentNode.appendChild(headerChildren[i]);
+                        }
                     }
                 }
             },
             headerChildren: {
                 get: function() {
-                    return document.querySelectorAll('mkt-header-child');
+                    return this.parentNode.querySelectorAll(
+                        'mkt-header-child');
                 }
             },
-            hideHeaderChildren: {
-                value: function() {
-                    var headerChildren = this.headerChildren;
+            statusElement: {
+                get: function() {
+                    return this.parentNode;
+                }
+            },
+            toggleChildren: {
+                value: function(arg) {
+                    // If arg is false, toggle all children off.
+                    // Else arg is an ID, toggle only that child on.
+                    var root = this;
+                    var headerChildren = root.headerChildren;
+
+                    var showingChild = false;
                     for (var i = 0; i < headerChildren.length; i++) {
-                        headerChildren[i].toggle(false);
+                        var child = headerChildren[i];
+
+                        if (child.id == arg) {
+                            child.toggle();
+
+                            if (child.visible) {
```

Comment:
But it's wrapped in `child.id == arg`. Shouldn't this only execute once?

Also seems like `child.id == arg` should be a `===`.